tests/unit/modules/test_virt.py

```diff
@@ -1120,6 +1121,54 @@ def test_init(self):
                 self.assertFalse('<interface' in definition)
                 self.assertFalse('<disk' in definition)
 
+                # Ensure the init() function allows creating VM without NIC and
+                # disk but with boot parameters.
+
+                defineMock.reset_mock()
+                mock_run.reset_mock()
+                boot = {
+                    'kernel': '/root/f8-i386-vmlinuz',
+                    'initrd': '/root/f8-i386-initrd',
+                    'cmdline':
+                        'console=ttyS0 ks=http://example.com/f8-i386/os/'
+                }
+                retval = virt.init('test vm boot params',
+                                   2,
+                                   1234,
+                                   nic=None,
+                                   disk=None,
+                                   seed=False,
+                                   start=False,
+                                   boot=boot)
+                definition = defineMock.call_args_list[0][0][0]
+                self.assertEqual('<kernel' in definition, True)
+                self.assertEqual('<initrd' in definition, True)
+                self.assertEqual('<cmdline' in definition, True)
+                self.assertEqual(retval, True)
+
+                # Verify that remote paths are downloaded and the xml has been
+                # modified
```

Comment:
You will need to mock the urlopen call for that test case. See https://github.com/saltstack/salt/blob/master/tests/unit/modules/test_nginx.py#L35 for an example of it. You could move the whole test in a context like this one:

```python
mock_response = MagicMock()
mock_response.read= MagicMock(return_value="filecontent")
with patch('salt.ext.six.moves.urllib.request.urlopen', MagicMock(return_value=mock_response)):
    defineMock.reset_mock()

Note that you will also need to replace the `'/var/lib/libvirt/saltinst'` with a temp folder in your tests since tests will not have access to this path for sure. Moving this value into a constant and wrapping the test in a context like this one would do the trick:

```python
cache_dir = tempfile.mkdtemp()                                                                                                                               
with patch.dict(virt.__dict__, {'CACHE_DIR': cache_dir}):
    # the whole test block
shutil.rmtree(cache_dir)
```

Reply:
Changes are in the next commit